### ISSUE TITLE
Stuttering in-game with one gamescope instance nested inside another

### ISSUE THREAD

**Issue by isaboll1** [Plagman/gamescope#452]
In utilizing one gamescope instance launched for a specific game, within another gamescope instance that represents the base compositor, I've noticed that doing so can lead to stuttering within the game. I noticed this when utilizing gamescope with steam integration (`gamescope -e -- steam -tenfoot -steamos`), and using steam launch commands to force different games to launch with different gamescope properties (such as forcing some games to launch with nearest neighbor filtering, or utilizing the `-e` option for games that have proton7 or experimental enabled, etc).

**Comment by Joshua-Ashton** [comment]
I don't really think nesting gamescope is a good idea to start with. We have a separate solution in gamescope (used on Deck) for isolating games like this, which is the multiple xwayland system.

To use it in the non-Deck UI would be just `gamescope --xwayland-count 2 ...` then `DISPLAY=:1 %command%` for the game, or whatever the x session + 1 of the base gamescope display is.

**Comment by isaboll1** [comment]
I see. With that method, is it possible to still change the properties of the second gamescope display with the command line options, to force the game to launch with a different max resolution, or forcing it to launch with a specific filtering/scaling method, etc?

**Comment by isaboll1** [comment]
After some further testing launching games without another gamescope instance, I noticed that there is still a bit of stuttering, and that it doesn't quite seem to be caused by nested instances (although that does make it worse). I tested with Red Dead Redemption, and for added context my system has an AMD RX 6600, and an AMD Ryzen 7 5700G. 

In utilizing steam with gamescope integration, and calling`export vblank_mode=1` and `export MESA_VK_WSI_PRESENT_MODE=fifo` beforehand, stuttering is gone in Red Dead if in-game, the vsync option is turned off.
In most other games such as Dragon Ball FighterZ or Guilty Gear XRD Rev2 that also helps in reducing the stutter, but it's not a panacea as other applications still exhibit it.
